src/mpi_globals.F90

```diff
@@ -52,6 +52,43 @@ subroutine negf_mpi_init(energyComm, ioProc)
 
     end subroutine negf_mpi_init
 
+    ! Initialize a 2D cartesian grid
+    subroutine negf_cart_init(inComm, nk, cartComm, energyComm, kComm)
+      type(mpifx_comm), intent(in) :: inComm
+      integer, intent(in) :: nk
+      type(mpifx_comm), intent(out) :: energyComm
+      type(mpifx_comm), intent(out) :: kComm
+
+      type(mpifx_comm) :: cartComm
```

Comment:
is this an output parameter as well, or you were intending to pass only energyCommand kComm? If it's a parameter, it needs `intent` and the declaration should be moved above between nk and energyComm